RELEASES.md

```diff
@@ -114,17 +114,15 @@ deadline of 3 weeks.
     release, please also make sure there's a stable maintainer signed up for next quarter,
     and the deadline for the next release is documented in the release schedule.
   * Get a review and merge.
-* Create a pull request with the commit created by the project tool and wait for tests to
-  pass on [main](https://dev.azure.com/cncf/envoy/_build).
-* Create a [tagged release](https://github.com/envoyproxy/envoy/releases). The release should
-  start with "v" and be followed by the version number. E.g., "v1.6.0". **This must match the
-  [VERSION](VERSION).**
+* Create a pull request with the commit created by the project tool and **wait for tests to
+  pass**.
+* Once the tests have passed, and the PR has landed, CI will automatically create the tagged release.
 * From the envoy [landing page](https://github.com/envoyproxy/envoy) use the branch drop-down to create a branch
   from the tagged release, e.g. "release/v1.6". It will be used for the
```

Comment:
maybe this needs to emphasize that the branch name only uses the minor version (not patch ie `v1.6.0`)